tests/rptest/clients/python_librdkafka.py

```diff
@@ -24,9 +24,9 @@ class PythonLibrdkafka:
     def __init__(self,
                  redpanda,
                  *,
-                 username=None,
-                 password=None,
-                 algorithm=None,
+                 username: str | None = None,
```

Comment:
When I looked into it, the this is the recommended syntax for PEP-484 when it is available (originally, it wasn't, only `Union`, `Optioinal` was). I think the reasoning is clearer if you compare `Foo | Bar` to `Union[Foo, Bar]` and then discouraging `Optional` just follows from that (though the terseness benefit is less obvious for the `Optional` case.

Here's one thread though not the one I read:

https://discuss.python.org/t/clarification-for-pep-604-is-foo-int-none-to-replace-all-use-of-foo-optional-int/26945/42

It has argument both ways though the consensus among the core dev/typing folks themselves seems to be to prefer the newer `Foo | None`.

Reply:
Cool, makes sense. I naturally reach for optional because it feels familiar, but it's _not_ an actual monad. I think "you'll need this to take advantage of python typing anyway" is as good an argument as any.